include/cantera/thermo/PlasmaPhase.h

```diff
@@ -0,0 +1,146 @@
+ * Journal of Plasma Physics 86.2 (2020).
+ * doi: https://doi.org/10.1017/S0022377820000161
+ *
+ * [3] G. J. M. Hagelaar and L. C. Pitchford
+ * "Solving the Boltzmann equation to obtain electron transport
+ * coefficients and rate coefficients for fluid models."
+ * Plasma Sources Science and Technology 14.4 (2005): 722.
+ * doi: https://doi.org/10.1088/0963-0252/14/4/011
+ *
+ * [4] A. Luque, "BOLOS: An open source solver for the Boltzmann equation,"
+ * https://github.com/aluque/bolos.
+ *
+ * @todo Implement electron Boltzmann equation solver to solve EEDF.
+ * @ingroup phase
+ */
+class PlasmaPhase: public IdealGasPhase
+{
+public:
+    //! Construct and initialize an PlasmaPhase ThermoPhase object
+    //! directly from an ASCII input file. The constructor initializes the electron
+    //! energy distribution to be Druyvesteyn distribution (m_x = 2.0). The initial
+    //! electron energy grid is set to a linear space which starts at 0.01 eV and ends
+    //! at 1 eV with 1000 points.
+    /*!
+     * @param inputFile Name of the input file containing the phase definition
+     *                  to set up the object. If blank, an empty phase will be
+     *                  created.
+     * @param id        ID of the phase in the input file. Defaults to the
+     *                  empty string.
+     */
+    explicit PlasmaPhase(const std::string& inputFile="",
+                         const std::string& id="");
+
+    virtual std::string type() const {
+        return "Plasma";
+    }
+
+    //! Set electron energy levels.
+    //! @param levels The vector of electron energy levels. [eV]
+    void setElectronEnergyLevels(const vector_fp& levels);
+
+    //! Get electron energy levels.
+    //! @param levels The vector of electron energy levels. [eV]
+    void getElectronEnergyLevels(vector_fp& levels) const;
+
+    //! Set electron energy distribution.
+    //! @param levels The vector of electron energy levels. [eV]
+    //! @param distrb The vector of electron energy distribution.
+    void setElectronEnergyDistribution(const vector_fp& levels, const vector_fp& distrb);
+
+    //! Get electron energy distribution.
+    //! @param distrb The vector of electron energy distribution.
+    void getElectronEnergyDistribution(vector_fp& distrb) const;
+
+    //! Update electron energy distribution.
+    void updateIsotropicElectronEnergyDistribution();
+
+    //! Set the internally stored electron temperature of the phase (K).
+    //! @param Te Electron temperature in Kelvin
+    virtual void setElectronTemperature(const double Te);
+
+    //! Mean electron energy [eV]
+    double meanElectronEnergy() {
+        return m_meanElectronEnergy;
+    }
+
+protected:
+    // Electron energy order in the exponential term
+    double m_x;
+
+    //! Number of points of electron energy grid
+    size_t m_nPoints;
+
+    //! electron energy grid
+    Eigen::VectorXd m_electronEnergyLevels;
```

Comment:
Other than allowing for 'economic' code, is there a reason to use `Eigen::VectorXd` rather than `vector_fp` here? Are there speed penalties? (this is mostly a question, not a suggestion).

Reply:
I found that it is much more convenient to use Eigen::VectorXd for matrix operation. In the future, it will be used for solving Boltzmann equation.